src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -113,56 +113,100 @@ impl<'a> JobQueue<'a> {
     pub fn execute(&mut self, cx: &mut Context) -> CargoResult<()> {
         let _p = profile::start("executing the job graph");
 
+        // We need to give a handle to the send half of our message queue to the
+        // jobserver helper thrad. Unfortunately though we need the handle to be
+        // `'static` as that's typically what's required when spawning a
+        // thread!
+        //
+        // To work around this we transmute the `Sender` to a static lifetime.
+        // we're only sending "longer living" messages and we should also
+        // destroy all references to the channel before this function exits as
+        // the destructor for the `helper` object will ensure the associated
+        // thread i sno longer running.
```

Comment:
Interesting, looks like this transmute is always safe, because `Sender` is contravariant.